### PR TITLE
Enable/disable pre-execution feature via configuration

### PR DESCRIPTION
**Pull request by yuliasherman**
Control pre-execution feature availability via a configuration file.

https://jira.eng.vmware.com/browse/BC-1808

### CODE REVIEW

File: bftengine/include/bftengine/ReplicaConfig.hpp [hunk 1]
```diff
@@ -66,6 +66,9 @@ struct ReplicaConfig {
   // a time interval in milliseconds, represents the timeout for automatically replacing the primary
   uint16_t autoPrimaryRotationTimerMillisec = 0;
 
+  // preExecutionFeatureEnabled=0/1; preExecutionFeatureEnabled=1 enables the pre-execution feature
+  uint8_t preExecutionFeatureEnabled = 0;
```
Comment (teoparvanov) [inline bftengine/include/bftengine/ReplicaConfig.hpp]: why not `bool` ?

Comment (yuliasherman) [reply bftengine/include/bftengine/ReplicaConfig.hpp]: Do we like to serialize booleans?

Comment (yuliasherman) [reply bftengine/include/bftengine/ReplicaConfig.hpp]: Iknow that we do that, but it is a bad practice.

Comment (teoparvanov) [reply bftengine/include/bftengine/ReplicaConfig.hpp]: I'd prefer to be consistent with the rest of the code - and do an eventual clean-up separately.

Comment (yuliasherman) [reply bftengine/include/bftengine/ReplicaConfig.hpp]: Hm. OK.

**Pull request merged**